### PR TITLE
Add Github actions for colcon build

### PR DESCRIPTION
**Pull request by Ryanf55**
# Purpose

Add a Github workflow to use Github actions to compile the repository. This replaces CircleCI.

# Ticket

Solves #415

### CODE REVIEW

File: .github/workflows/colcon.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Colcon Build Test
+on:
+  push:
+    # branches:
+    # - 'rolling'
+    # - 'iron'
+    # - 'humble'
```
Comment (wep21) [inline .github/workflows/colcon.yaml:7]: @Ryanf55 Why do you comment out branches?

Comment (Ryanf55) [reply .github/workflows/colcon.yaml]: Ah, unintentional just to make this easier to iterate on without having a PR. I've dropped that commit.

Now, CI only runs this job if
* It's a PR
* It's a commit on any of those three branches (which happens just after a merge).

File: .github/workflows/colcon.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    branches:
+    - '*'
+jobs:
+  build-test:
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {rosdistro: 'rolling', container: 'ros:rolling'}
```
Comment (wep21) [inline .github/workflows/colcon.yaml:18]: Can we use rolling when the target branch is not rolling?

Comment (Ryanf55) [reply .github/workflows/colcon.yaml:18]: I'll manually change this on the backport.

**Review by wep21 — APPROVED** [review]
LGTM

**Comment by Ryanf55** [comment]
@mergifyio backport iron

**Comment by Ryanf55** [comment]
@mergifyio backport humble

**Pull request merged**